node/contract.go

```diff
@@ -36,12 +36,12 @@ func (node *Node) AddStakingContractToPendingTransactions() {
 	dataEnc := common.FromHex(StakingContractBinary)
 	// Unsigned transaction to avoid the case of transaction address.
 	mycontracttx, _ := types.SignTx(types.NewContractCreation(uint64(0), node.Consensus.ShardID, contractFunds, params.TxGasContractCreation*10, nil, dataEnc), types.HomesteadSigner{}, priKey)
-	node.ContractAddresses = append(node.ContractAddresses, crypto.CreateAddress(contractAddress, uint64(0)))
+	node.StakingContractAddress = crypto.CreateAddress(contractAddress, uint64(0))
```

Comment:
this is not right

Reply:
you need to get deployed contract address.

Reply:
Sorry, this hack logic is correct by this https://github.com/harmony-one/go-ethereum/blob/38cce9ac333d674616047be14c270d7cfbd43641/core/state_processor.go#L119

but we had 2 assumptions here
1/ nonce is 0
2/ you assumed  the address only creates one contract.

let put a separated function for this.

Reply:
Done.